frmts/ecw/ecwcreatecopy.cpp

```diff
@@ -211,24 +217,67 @@ CNCSError GDALECWCompressor::WriteReadLine( UINT32 nNextLine,
     for( iBand = 0; iBand < sFileInfo.nBands; iBand++ )
         panBandMap[iBand] = iBand+1;
 
-    GByte *pabyLineBuf =
-        (GByte *) CPLMalloc( sFileInfo.nSizeX * sFileInfo.nBands
-                             * nWordSize );
+    if( m_poSrcDS == nullptr || m_poSrcDS->GetRasterBand(1) == nullptr )
+    {
+        return GetCNCSError(NCS_FILEIO_ERROR);
+    }
+    if( m_nSwathLines <= 0 )
+    {
+        int nBlockX;
+        constexpr int MIN_SWATH_LINES = 256;
+        m_poSrcDS->GetRasterBand(1)->GetBlockSize( &nBlockX, &m_nSwathLines );
+        m_nSwathLines = std::min(MIN_SWATH_LINES, m_nSwathLines);
```

Comment:
Unfortunately std::min conflicts with NOMINMAX macro. We need to downgrade to "C" macro.